cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -225,6 +247,38 @@ class string_scalar : public scalar {
    , _data(string.data(), string.size(), stream, mr)
   {}
 
+  /**
+   * @brief Construct a new string scalar object from string_view
+   * Note that this function copies the data pointed by string_view.
+   * 
+   * @param source string_view pointing string value to copy
+   * @param is_valid Whether the value held by the scalar is valid
+   * @param stream The CUDA stream to do the allocation in
+   * @param mr The memory resource to use for allocation
+   */
+  string_scalar(value_type const& source, bool is_valid = true, 
+      cudaStream_t stream = 0,
+      rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource())
+   : scalar(data_type(STRING), is_valid)
+   , _data(source.data(), source.size_bytes(), stream, mr)
+  {}
+
+  /**
+   * @brief Construct a new string scalar object from string_view in device memory
+   * Note that this function copies the data pointed by string_view.
+   * 
+   * @param data device_scalar string_view pointing string value to copy
+   * @param is_valid Whether the value held by the scalar is valid
+   * @param stream The CUDA stream to do the allocation in
+   * @param mr The memory resource to use for allocation
+   */
+  string_scalar(rmm::device_scalar<value_type>&& data, bool is_valid = true, cudaStream_t stream = 0,
+      rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource())
+   : string_scalar(data.value(), is_valid, stream, mr)
+  {
+    data.~device_scalar();
```

Comment:
I am trying to copy the string pointed by `string_view`, and destruct the `data` object since this is a move constructor. 
This line causes memory cudaFree Invalid device pointer error.
@jrhemstad @rapidsai/cudf-cpp-codeowners  @harrism @devavret Can someone point me what's wrong here?

Reply:
Why is data a `device_scalar`? String should be in `device_buffer` right?

Reply:
Min/Max operation is performed on iterator of `string_view`s, So the reduction result is intermediate non-owning `string_view`.  This non-owning `string_view` is converted to owning `string_scalar` (which contains the `device_buffer`).  So this intermediate result is allocated by `device_scalar<string_view>`.